R/clean_up_text_variables.R

```diff
@@ -0,0 +1,30 @@
+#' Clean Up Text Variables
+#'
+#' @param data data containing string variables needing cleaned up
+#' @param string string variable
+#'
+#' @return data with cleaned up string variables
+#' @export
+clean_up_free_text <- function(data, string, remove_punct = TRUE) {
+  if (remove_punct == TRUE) {
+    data <- data %>%
+      dplyr::mutate(
+        # deal with capitalisation of CH names
+        {{ string }} := stringr::str_to_title({{ string }}),
+        # deal with whitespace at start and end and witihin
+        {{ string }} := stringr::str_trim({{ string }}, side = "both"),
+        {{ string }} := stringr::str_squish({{ string }}),
+        # deal with punctuation in the CH names
+        {{ string }} := stringr::str_replace_all({{ string }}, "[[:punct:]]", " ")
+      )
+  } else {
+    data <- data %>%
+      dplyr::mutate(
+        # deal with capitalisation of CH names
+        {{ string }} := stringr::str_to_title({{ string }}),
+        # deal with whitespace at start and end and witihin
+        {{ string }} := stringr::str_trim({{ string }}, side = "both"),
+        {{ string }} := stringr::str_squish({{ string }})
+      )
+  }
+}
```

Comment:
Some changes:

- Changes to the doc comments
- Changes to the code comments
- Take a string instead of a dataframe (returns the cleaned string)
- Make the case_to specifiable as an argument (title case is the default)

```suggestion
#' Clean up a free text string
#'
#' @description Take a messy string and clean it up by converting it to title
#' case, removing any superfluous whitespace and optionally removing
#' any punctuation. The use case is to make text style uniform to aid with
#' matching.
#'
#' @param string string variable
#' @param case_to the case to convert the string to
#' @param remove_punct Should any punctuation be removed?
#' (default `TRUE`)
#'
#' @return The cleaned string
#' @export
#' @examples
#' clean_up_free_text("hiwSDS SD. h")
clean_up_free_text <- function(string, case_to = c("upper", "lower", "sentence", "title", "none"),
                               remove_punct = TRUE) {

  if (missing(case_to)) case_to <- "title"

  case_to <- match.arg(case_to)

  cleaned_string <-
    # Deal with whitespace at start and end and within
    stringr::str_squish(stringr::str_trim(string, side = "both"))

  # Remove any punctuation (optionally)
  if (remove_punct) {
    # deal with punctuation in the CH names
    cleaned_string <- stringr::str_replace_all(cleaned_string, "[[:punct:]]", " ")
  }

  # Make the case uniform
  cleaned_string <- dplyr::case_when(
    case_to == "lower" ~ stringr::str_to_lower(cleaned_string),
    case_to == "upper" ~ stringr::str_to_upper(cleaned_string),
    case_to == "sentence" ~ stringr::str_to_sentence(cleaned_string),
    case_to == "title" ~ stringr::str_to_title(cleaned_string),
    case_to == "none" ~ cleaned_string
  )

  return(cleaned_string)
}
```